src/main/java/hudson/tasks/junit/JUnitParser.java

```diff
@@ -116,9 +126,9 @@ public TestResult invoke(File ws, VirtualChannel channel) throws IOException {
                 throw new AbortException(Messages.JUnitResultArchiver_NoTestReportFound());
             }
 
-            TestResult result = new TestResult(buildTime + (nowSlave - nowMaster), ds, keepLongStdio);
+            TestResult result = new TestResult(buildTime + (nowSlave - nowMaster), ds, keepLongStdio, listener);
```

Comment:
Invoke is being executed on the slave's side. Most probably TaskListener won't behave as you expect